### PR TITLE
implement job persistance

### PR DESCRIPTION
**Pull request by mxdev88**
Hello,

Could someone pls review this PR about #12?

Thank you

### CODE REVIEW

**Comment by mxdev88** [comment]
Anyone to take a first look? @Digenis?

**Comment by Digenis** [comment]
Hi, @mxdev88,
thank you for your contribution.

I don't have much time to maintain scrapyd nowadays.
I only took a quick look.

The travis error can be ignored, it's #359.
Overall it seems cleanly implemented
and I guess it will not complicate merging the queues.

**Comment by mxdev88** [comment]
Hi @Digenis,
Great - thanks for the feedback. Let me know if anything else needed prior to merging.

**Comment by sluceno** [comment]
Not an expert on python but it looks good to me. 

I will take this as an example for doing similar thing with the Queue so we can have different backends for queueing jobs

**Comment by skykery** [comment]
Hope this feature will be merged someday.

**Comment by markkdev** [comment]
Would be awesome to have this feature

**Comment by mxdev88** [comment]
@Digenis do you think this one could make 1.3 alpha release mentioned in #389 also?

**Comment by Digenis** [comment]
I added the 1.3 milestone although I haven't estimated the time to review yet.

I see there is merge conflict but an easy to resolve.

You'll probably want to rebase on master after #398 to get a working CI.

**Comment by Digenis** [comment]
@mxdev88 you can now rebase on master.
The conflicts are only precautions
because of the proximity of the changed lines.

**Comment by mxdev88** [comment]
Hi @Digenis should be ok to merge now. thks

**Comment by polajenko** [comment]
When can this feature be released?

**Comment by pawelmhm** [comment]
This looks really useful, there are unit tests, can you merge recent master branch here and rebuild? Also: finished_to_keep setting seems not documented, worth adding to docs.

File: docs/config.rst [hunk 1]
```diff
@@ -152,6 +152,17 @@ Scrapyd includes an interface with a website to provide simple monitoring
 and access to the application's webresources.
 This setting must provide the root class of the twisted web resource.
 
+jobstorage
+-------
+
+A class that stores finished jobs. There are 2 implementations provided:
+
+* ``MemoryJobStorage`` (default) jobs are stored and memory and lost when the daemon is restarted
```
Comment (pawelmhm) [inline docs/config.rst:160]: default is actually:

scrapyd.jobstorage.MemoryJobStorage

let's put full names here:

scrapyd.jobstorage.MemoryJobStorage
and scrapyd.jobstorage.SqliteJobStorage

**Comment by pawelmhm** [comment]
I integrated with current master and fixe docs here:https://github.com/scrapy/scrapyd/pull/418

**Comment by pawelmhm** [comment]
related PR merged, commit in master now

**Pull request closed without merging**